plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/data/onboarding-context.tsx

```diff
@@ -203,52 +229,139 @@ export const OnboardingProvider: React.FC< {
 		[ getStepByKey, history, urlStrategy ]
 	);
 
-	// Find the first incomplete step with completed dependencies
-	const currentStep = allSteps.find(
-		( step ) =>
-			step.status !== 'completed' &&
-			areStepDependenciesCompleted( step, allSteps )
-	);
+	const findFirstIncompleteStep = useCallback(
+		(
+			steps: WooPaymentsProviderOnboardingStep[],
+			allStepsCollection: WooPaymentsProviderOnboardingStep[],
+			parentStep?: WooPaymentsProviderOnboardingStep
+		): WooPaymentsProviderOnboardingStep | undefined => {
+			for ( const [ index, step ] of steps.entries() ) {
+				// Special completion check for frontend steps.
+				if ( step.type === 'frontend' ) {
+					let isCompleted = false;
+					if ( parentStep ) {
+						// Rule 2: A frontend sub-step is completed if its parent is, or if the next backend sibling is active/done.
```

Comment:
Add this logic here to make sure frontend steps aren't "sabotaging" the progress. If the next backend sibling is completed, then the frontend step is also completed.